src/actions.js

```diff
@@ -0,0 +1,48 @@
+const add = (tasks, input) => {
+  const task = {
+    description: input,
+    completed: false,
+    index: tasks.length + 1,
+  };
+  tasks.push(task);
+  return tasks;
+};
+
+const shiftIndex = (tasks, taskIndex) => {
+  tasks = tasks.map((task) => {
+    if (task.index > taskIndex) {
+      task.index -= 1;
+    }
+    return task;
+  });
+  return tasks;
+};
+
+const remove = (tasks, task) => {
+  const taskIndex = tasks.findIndex((item) => (item.index === task.index));
+  tasks = tasks.filter((item) => (item.index !== task.index));
+  tasks = shiftIndex(tasks, taskIndex);
+  return tasks;
+};
+
+const edit = (tasks, task, desc) => {
+  tasks = tasks.map((item) => {
+    if ((item.index === task.index) && (item.description === task.description)) {
+      item.description = desc;
+    }
+    return item;
+  });
+  return tasks;
+};
```

Comment:
- [ ] It would be better when you update the task you can see the change without refreshing the page. This will help you to improve the function of the page. Currently, you must refresh the page to see the update. Kindly fix this minor issue. 
🛠️ 👍